### ISSUE TITLE
`BUG` TFT Upload with esphome 4.2.2 not possible but works with 4.2.1

### ISSUE THREAD

**Issue by MichaelHeimann** [Blackymas/NSPanel_HA_Blueprint#1659]
### TFT Version

4.2.2

### ESPHome Version

4.2.2

### Blueprint Version

4.2.2

### Panel Model

NSPanel EU Model

### What is the bug?

trying to update the tft from esphome 4.2.2 

### Steps to Reproduce

1. install esphome 4.2.2
2. push "Update TFT display" in HA
3. read only failed attempts in esphome log
sidenote1: tft was stock before.
sidenote2: installing/downgrading to esphome 4.2.1 fixed the issue and the tft updated fine (tried 5 times with 4.2.2)

[tft-upload-fail_4_2_2_logs_nstoilette_logs.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13955606/tft-upload-fail_4_2_2_logs_nstoilette_logs.txt)

### Your panel's YAML

```yaml
substitutions:

###### CHANGE ME START ######

  device_name: "nstoilette" 
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password

  nextion_update_url: "http://10.250.23.6:8123/local/nspanel_eu.tft" # URL to local tft File
#  nextion_update_url: "https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/main/nspanel_eu.tft" # URL to Github

  ##### addon-configuration #####
  ## addon_climate ##
  heater_relay: "1" # possible values: 1/2
  temp_units: "°C" #Temperatures in Fahrenheit
  temp_min: "16" #Min supported temperature is 40°F
  temp_max: "28" #Max supported temperature is 80°F
  temp_step: "0.5" #Temperature granularity is 1°F


##### CHANGE ME END #####

esphome:
  compile_process_limit: 2

##### DO NOT CHANGE ANYTHING! #####
api:
  encryption:
    key: "removed"

packages:
  ##### download esphome code from Github
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files:
      - nspanel_esphome.yaml # Core package
      # - nspanel_esphome_addon_climate_cool.yaml # activate for local climate (cooling) control
      - nspanel_esphome_addon_climate_heat.yaml # activate for local climate (heater) control
    refresh: 1s

esp32:
  framework:
    type: esp-idf

##### DO NOT CHANGE ANYTHING! #####
```

### ESPHome logs

```logs
esphome logs in attached tft-upload-fail_4_2_2_logs_nstoilette_logs.txt
```

**Comment by MichaelHeimann** [comment]
updating an existing 4.2.1 display to 4.2.2 did work with esphome 4.2.2, seems only to be related to a stock -> 4.2.2 issue.

**Comment by MichaelHeimann** [comment]
Just tried again with another stock nspanel and again the same issue. 4.2.2 tries I think 12 times with different baud rates and gives up then. 
4.2.1 works this time after try 6. Not perfect but obviously better than not working.
sidenote: also with 4.2.1 I need a second push on "update tft display" because the first time I get the system data error. the second time did work then (consistantly on 4 panels now).

**Comment by SeggersCoe** [comment]
Same here. Is it possible to downgrade to 4.21

**Comment by edwardtfn** [comment]
I've reverted the changes. It is in `dev`, but it would be included in the patch v4.2.3.

**Comment by MichaelHeimann** [comment]
> I've reverted the changes. It is in `dev`, but it will be included in the patch v4.2.3.

Just tried another stock panel with esphome from dev and it is uploading the tft with first attempt of pushing "update tft display" - attempt 7 - so basically like 2.4.1 - works again - thanks!

**Comment by edwardtfn** [comment]
Maybe I should change the routine a bit to move that try #7 a bit earlier in the sequence or try to identify when a TFT is not from our project to try that first.

Do you have a log including all those tries in a stock panel?

**Comment by MichaelHeimann** [comment]
> Maybe I should change the routine a bit to move that try 7 a bit earlier in the sequence or try to identify when a TFT is not from our project to try that first.
> 
> Do you have a log including all those tries in a stock panel?

will get one, still have some panels to do ;)

**Comment by edwardtfn** [comment]
So I will work in a sequence where the legacy upload is prioritized if TFT is not detected. Give me a few minutes, please.

**Comment by edwardtfn** [comment]
Done!
Could you please try `dev` and let me know your results?

**Comment by SeggersCoe** [comment]
same issue here. Nspanel brand new. flashed with newest version 4.2.2
[logs_blackymasschlafzimmer_logs.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13961795/logs_blackymasschlafzimmer_logs.txt)

**Comment by MichaelHeimann** [comment]
> Done! Could you please try `dev` and let me know your results?

Very nice! Upload starts at attempt 1:

[12:54:51][D][addon_upload_tft.script.report_upload_progress:203]: Attempt #1 at 115200 bps
[12:54:51][D][addon_upload_tft.script.upload_tft_v414:562]: Starting...
[12:54:51][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1059]: Nextion TFT upload requested
[12:54:51][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1060]:   url: http://10.250.23.6:8123/local/nspanel_eu.tft
[12:54:51][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1061]:   baud_rate: 115200
[12:54:51][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1074]: Setting Nextion protocol reparse mode to passive
[12:54:51][D][main:232]: Exit reparse
[12:54:51][D][addon_upload_tft.script.upload_tft_v414:573]: Wait 5 seconds
[12:54:56][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1114]: TFT file size: 7311536
[12:54:56][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1121]: File size check passed. Proceeding...
[12:54:56][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1126]: Updating Nextion
[12:54:56][D][addon_upload_tft.script.upload_tft.send_nextion_command:598]: Sending: whmi-wris 7311536,115200,1
[12:54:56][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1150]: Waiting for upgrade response
[12:54:57][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1155]: Upgrade response is [05]
[12:54:57][D][addon_upload_tft.script.upload_tft.upload_tft_esp_idf:1166]: Updating tft from "http://10.250.23.6:8123/local/nspanel_eu.tft" with a file size of 7311536, Heap Size 142076
[12:55:02][D][addon_upload_tft.script.upload_tft.upload_range_esp_idf_:1017]: Uploaded 0.06 %, remaining 7307440 bytes
[12:55:02][D][addon_upload_tft.script.upload_tft.upload_range_esp_idf_:1020]: recv_string [08.00.00.00.00 (5)]
[12:55:02][D][addon_upload_tft.script.upload_tft.upload_range_esp_idf_:1017]: Uploaded 0.11 %, remaining 7303344 bytes
[12:55:03][D][addon_upload_tft.script.upload_tft.upload_range_esp_idf_:1017]: Uploaded 0.17 %, remaining 7299248 bytes
[12:55:03][D][addon_upload_tft.script.upload_tft.upload_range_esp_idf_:1017]: Uploaded 0.22 %, remaining 7295152 bytes
[12:55:03][D][esp-idf:000]: I (127501) HTTP_CLIENT: Body received in fetch header state, 0x3ffdcea9, 91

[12:55:04][D][addon_upload_tft.script.upload_tft.upload_range_esp_idf_:1017]: Uploaded 0.28 %, remaining 7291056 bytes
[12:55:04][D][addon_upload_tft.script.upload_tft.upload_range_esp_idf_:1017]: Uploaded 0.34 %, remaining 7286960 bytes

**Comment by MichaelHeimann** [comment]
Hmm, bad news:
It didn't upload the full tft at once (as before) but now it can't copy the rest over. It seems it does but I'm stuck with System Data ERROR.
Theory:
Original try 7 can start the tft upload to overwrite a stock TFT.
Original try <7 was able to finish it in the second attempt.
Now try 1 starts but it cannot be finished since original try7 is executed first.

hope that makes sense.

Logs:
[tft_upload_completition_fail_logs_nsschlafzimmer_logs.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13963052/tft_upload_completition_fail_logs_nsschlafzimmer_logs.txt)

**Comment by edwardtfn** [comment]
I have double checked and all the waits are there, so we shouldn't have two paths in parallel.
This "System data ERROR" message can happens with any disturbance to the transfer flow and a retry should resolve that.

Basically, the system now have the same sequence as it was in v4.2.1, however, if a TFT haven't reported it's Id to ESPHome (which all of our TFTs after v4.0 will do), the system will try the legacy transfer engine before start the regular sequence (which also includes the legacy again at attempt 7, as I haven't removed that).

**Comment by edwardtfn** [comment]
I will revert back this change to start that one first. Please let me know the results.

**Comment by edwardtfn** [comment]
Done!
So now the upload script should be the same as v4.2.1

**Comment by MichaelHeimann** [comment]
> This "System data ERROR" message can happens with any disturbance to the transfer flow and a retry should resolve that.

This happens to me relyable with 6 panels now. all updated in the 4.2.1 und 4.2.2 code timeframe and all get interrupted at about 8x% - then continue when I press upload tft again in HA - then it's completed.

the network RSSI of the panels varies from 45 to 70 and still the experience is similar so I think this disturbance in transfer flow is some kind of buffer underrun on the device or other interruptions the esp doesn't understand and just continues (because the precentage always counts to 100 but when it resumes it starts at 8x% and finishes.

I'll try dev tree with another panel now and report back

**Comment by luixal** [comment]
Hi, newbie here who just hit this issue :)

In order to try this, update or use a previous version... are there any docs about how to?

I was thinking about creating a new automation using a link to the blueprint from another branch, but I'm unsure if that would do the trick.

Thanks.

**Comment by MichaelHeimann** [comment]
> Can I just change the url in packages > remote_package section to the url of the branch?
I had the same question, and found out that you can just edit the yaml of the esp device

   ref: main
->
    ref: dev

changes from main to dev

**Comment by luixal** [comment]
Tried changing ref to `dev`, flashing and updating tft display... same error as before:

```[E][nextion.upload.idf:228]: Preparation for tft update failed 0 ""```

until attempt #7 (this didn't happened before, as you said) now it seems to be working.

I think you already provided the full log of attempts so I'm not including it here.

Thanks! This is my first nspanel and I was completely lost until finding this issue.

**Comment by luixal** [comment]
After flashing, I'm stuck at the ```"System Data ERROR!"``` screen.

```
[15:58:08][D][script.setup_sequence:2661]: Starting Nextion setup sequence
[15:58:08][D][script.setup_sequence:2662]: Fetching page id
[15:58:08][E][nextion:525]: ERROR: Received numeric return but the queue is empty
[15:58:13][D][script.setup_sequence:2670]: Fetching charset
[15:58:13][W][nextion:405]: Nextion reported variable name invalid!
[15:58:18][D][script.setup_sequence:2678]: Fetching display mode
[15:58:18][W][nextion:405]: Nextion reported variable name invalid!
[15:58:19][I][addon_upload_tft.script.report_settings:189]: TFT URL: http://192.168.68.53:8123/local/nspanel_eu.tft
[15:58:19][I][addon_upload_tft.script.report_settings:190]: Substitutions:
[15:58:19][I][addon_upload_tft.script.report_settings:191]:   nextion_update_url:        http://192.168.68.53:8123/local/nspanel_eu.tft
[15:58:19][I][addon_upload_tft.script.report_settings:192]:   nextion_update_base_url:   https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/
[15:58:19][I][addon_upload_tft.script.report_settings:193]:   upload_tft_chunk_size_max: 32768
[15:58:19][I][addon_upload_tft.script.report_settings:194]:   upload_tft_baud_rate:      921600
[15:58:23][E][script.setup_sequence:2805]: A non-compatible TFT is installed
[15:58:23][E][script.setup_sequence:2806]: Display mode: nan
```

After pressing the "Update TFT Display" again (although I selected the remote file), it showed this and I guess it ended writting the tft file:

```
[15:59:30][E][nextion:285]: Nextion queue is empty!
[15:59:32][D][esp32.preferences:114]: Saving 2 preferences to flash...
[15:59:32][D][esp32.preferences:143]: Saving 2 preferences to flash: 0 cached, 2 written, 0 failed
[15:59:32][D][main:1203]: Attempting to upload TFT
[15:59:32][D][addon_upload_tft.script.report_upload_progress:203]: Attempt #1 at 921600 bps
[15:59:32][D][addon_upload_tft.script.nextion_upload:153]: Waiting for empty UART and Nextion queues
[15:59:34][D][addon_upload_tft.script.nextion_upload:162]: Starting TFT upload...
[15:59:34][D][nextion.upload.idf:133]: Nextion TFT upload requested
[15:59:34][D][nextion.upload.idf:134]: url: http://192.168.68.53:8123/local/nspanel_eu.tft
[15:59:34][D][nextion.upload.idf:183]: TFT file size: 7311536
[15:59:34][D][nextion.upload.idf:195]: Updating Nextion
[15:59:35][D][nextion.upload.idf:223]: Upgrade response is [05]
[15:59:35][D][nextion.upload.idf:234]: Updating tft from "http://192.168.68.53:8123/local/nspanel_eu.tft" with a file size of 7311536, Heap Size 149144
[15:59:35][D][esp-idf:000]: I (96986) HTTP_CLIENT: Body received in fetch header state, 0x3ffdbc95, 99

[15:59:40][D][nextion.upload.idf:090]: Uploaded 0.06 %, remaining 7307440 bytes
[15:59:40][D][nextion.upload.idf:094]: recv_string [08.00.00.61.00 (5)]
[15:59:40][I][nextion.upload.idf:103]: Nextion reported new range 6356992
[15:59:40][D][nextion.upload.idf:090]: Uploaded 87.00 %, remaining 950448 bytes
[15:59:40][D][nextion.upload.idf:090]: Uploaded 87.06 %, remaining 946352 bytes
[15:59:40][D][nextion.upload.idf:090]: Uploaded 87.11 %, remaining 942256 bytes
[15:59:40][D][nextion.upload.idf:090]: Uploaded 87.17 %, remaining 938160 bytes
[15:59:40][D][nextion.upload.idf:090]: Uploaded 87.22 %, remaining 934064 bytes
```

Quite weird.

**Comment by MichaelHeimann** [comment]
So, I've tried a new panel with dev and it's now again the old behaviour - try 1 to 6 fail and 7 starts initially and then I need to restart to get the rest written but that works at try 1


[tft_upload_logs_dev_try_1_to_6_fail.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13965390/tft_upload_logs_dev_try_1_to_6_fail.txt)

**Comment by MichaelHeimann** [comment]
> Quite weird.

Yeah, especially since the 87% restart mark is exactly where it picks up for me, too.

**Comment by bluemaster27** [comment]
I updated an existing 4.2.1 panel with 4.2.2, now HA gives me this notification:

NSPanel component version mismatch
There's a version mismatch between NSPanel_HA_Blueprint components. Please update all the components to the same version.

Panel: nspanel
TFT: 4.1.4
ESPHome: 4.2.2
Blueprint: 4.2.2 (expected 4.2.2)

Is it related to this bug, or did I forgot something?

**Comment by whitexichigo** [comment]
> I updated an existing 4.2.1 panel with 4.2.2, now HA gives me this notification:
> 
> NSPanel component version mismatch There's a version mismatch between NSPanel_HA_Blueprint components. Please update all the components to the same version.
> 
> Panel: nspanel TFT: 4.1.4 ESPHome: 4.2.2 Blueprint: 4.2.2 (expected 4.2.2)
> 
> Now the NSpanel screen stays black. Is it related to this bug, or did I forgot something?

i have the exact same problem

**Comment by robbiedehand** [comment]
@bluemaster27 @whitexichigo from the device menu in ha open that device and klik update tft
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/88730917/0de0d235-1b20-45fc-a454-5e46a2166bb8), that should resolve it.
Mind you I had the same error and I downloaded the current tft file renamed the old one and only after that I clicked update tft and that seemed to work for me

**Comment by edwardtfn** [comment]
```
Panel: nspanel
TFT: 4.1.4
ESPHome: 4.2.2
Blueprint: 4.2.2 (expected 4.2.2)
```

You need all the parts to be in the same version (changes on the last number is ok), and this shows your TFT is in an earlier version  (v4.1.x), compared to the other two parts (ESPHome and Blueprint) which are on v4.2.x.

So, focus on updating the TFT. If you get some error message in your screen, that's not a problem, it's just about try again. We have a whole guide for this TFT Upload process: [Troubleshooting TFT transfer](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/tft_upload.md)

If you are using `esp-idf` framework, just go to your device's page under Settings > Devices & Services > ESPHome and select the right model under "Upload TFT display - Model", wait a couple of seconds, then press "Upload TFT display".

If you are using `arduino` framework, then it's a bit more of work, as you should download the file from our [GitHub repository](https://github.com/Blackymas/NSPanel_HA_Blueprint/) (select the right one based on your model), save it in your computer, then upload it to a local web server (usually Home Assistant under `www` folder), and then set the `nextion_update_url` substitution to point to that file. As you already have v4.1.4 installed, you probably have done this before, so it's just about update the file in your local server.
After all that, then you press "Upload TFT display" (in your device's page under Settings > Devices & Services > ESPHome).

If you still having issue, please [create a new issue/bug](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/new/choose) and share your logs. We will be happy to help.

**Comment by SeggersCoe** [comment]
Thats my Response

[10:26:11][D][main:561]: Trying to upload TFT [10:26:11][D][addon_upload_tft.script.report_upload_progress:192]: Try #4 at 115200 bps [10:26:11][D][addon_upload_tft.script.nextion_upload:142]: Waiting for empty UART and Nextion queues [10:26:13][D][addon_upload_tft.script.nextion_upload:151]: Starting TFT upload... [10:26:13][D][nextion.upload.idf:133]: Nextion TFT upload requested [10:26:13][D][nextion.upload.idf:134]: url: https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/main/nspanel_eu.tft [10:26:14][D][nextion.upload.idf:183]: TFT file size: 7311536 [10:26:14][D][nextion.upload.idf:195]: Updating Nextion [10:26:16][D][nextion.upload.idf:223]: Upgrade response is [] [10:26:16][E][nextion.upload.idf:228]: Preparation for tft update failed 0 "" [10:26:16][D][nextion.upload.idf:259]: Restarting Nextion [10:26:17][W][component:214]: Component script took a long time for an operation (4.54 s). [10:26:17][W][component:215]: Components should block for at most 20-30ms. [10:26:17][D][addon_upload_tft.script.report_upload_progress:192]: Try #4 at 115200 bps failed! [10:26:22][D][script.nextion_status:1954]: Nextion status: [10:26:22][D][script.nextion_status:1955]: Is detected: True [10:26:22][D][script.nextion_status:1956]: Is setup: False [10:26:22][D][script.nextion_status:1957]: Queue size: 0 [10:26:22][D][addon_upload_tft.script.report_upload_progress:192]: Setting baud rate [10:26:24][D][main:561]: Trying to upload TFT [10:26:24][D][addon_upload_tft.script.report_upload_progress:192]: Try #5 at 115200 bps [10:26:24][D][addon_upload_tft.script.nextion_upload:142]: Waiting for empty UART and Nextion queues [10:26:26][D][addon_upload_tft.script.nextion_upload:151]: Starting TFT upload... [10:26:26][D][nextion.upload.idf:133]: Nextion TFT upload requested [10:26:26][D][nextion.upload.idf:134]: url: https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/main/nspanel_eu.tft [10:26:27][D][nextion.upload.idf:183]: TFT file size: 7311536 [10:26:27][D][nextion.upload.idf:195]: Updating Nextion [10:26:29][D][nextion.upload.idf:223]: Upgrade response is [] [10:26:29][E][nextion.upload.idf:228]: Preparation for tft update failed 0 "" [10:26:29][D][nextion.upload.idf:259]: Restarting Nextion [10:26:31][W][component:214]: Component script took a long time for an operation (4.55 s). [10:26:31][W][component:215]: Components should block for at most 20-30ms. [10:26:31][D][addon_upload_tft.script.report_upload_progress:192]: Try #5 at 115200 bps failed! [10:26:36][D][main:561]: Trying to upload TFT [10:26:36][D][addon_upload_tft.script.report_upload_progress:192]: Try #6 at 115200 bps [10:26:36][D][addon_upload_tft.script.nextion_upload:142]: Waiting for empty UART and Nextion queues [10:26:37][D][sensor:094]: 'ntc_source': Sending state 1.66000 V with 2 decimals of accuracy [10:26:37][D][resistance:039]: 'resistance_sensor' - Resistance 11336.6Ω [10:26:37][D][sensor:094]: 'resistance_sensor': Sending state 11336.58496 Ω with 1 decimals of accuracy [10:26:37][D][ntc:026]: 'blackymas3schlafzimmer Temperature' - Temperature: 22.2°C [10:26:37][D][sensor:094]: 'blackymas3schlafzimmer Temperature': Sending state 22.20327 °C with 1 decimals of accuracy [10:26:38][D][addon_upload_tft.script.nextion_upload:151]: Starting TFT upload... [10:26:38][D][nextion.upload.idf:133]: Nextion TFT upload requested [10:26:38][D][nextion.upload.idf:134]: url: https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/main/nspanel_eu.tft [10:26:39][D][nextion.upload.idf:183]: TFT file size: 7311536 [10:26:39][D][nextion.upload.idf:195]: Updating Nextion [10:26:41][D][nextion.upload.idf:223]: Upgrade response is [] [10:26:41][E][nextion.upload.idf:228]: Preparation for tft update failed 0 "" [10:26:41][D][nextion.upload.idf:259]: Restarting Nextion [10:26:43][W][component:214]: Component script took a long time for an operation (4.56 s). [10:26:43][W][component:215]: Components should block for at most 20-30ms. [10:26:43][D][addon_upload_tft.script.report_upload_progress:192]: Try #6 at 115200 bps failed! [10:26:48][D][script.nextion_status:1954]: Nextion status: [10:26:48][D][script.nextion_status:1955]: Is detected: True [10:26:48][D][script.nextion_status:1956]: Is setup: False [10:26:48][D][script.nextion_status:1957]: Queue size: 0 [10:26:48][D][addon_upload_tft.script.report_upload_progress:192]: Exiting reparse mode [10:26:48][D][main:1799]: Exit reparse [10:26:48][W][nextion:744]: Received unknown event from nextion: 0x55 [10:26:48][D][sensor:094]: 'blackymas3schlafzimmer RSSI': Sending state -54.00000 dBm with 0 decimals of accuracy [10:26:53][D][script.nextion_status:1954]: Nextion status: [10:26:53][D][script.nextion_status:1955]: Is detected: True [10:26:53][D][script.nextion_status:1956]: Is setup: False [10:26:53][D][script.nextion_status:1957]: Queue size: 0 [10:26:53][D][addon_upload_tft.script.report_upload_progress:192]: Setting baud rate [10:26:55][D][main:561]: Trying to upload TFT [10:26:55][D][addon_upload_tft.script.report_upload_progress:192]: Try #7 at 115200 bps [10:26:55][D][addon_upload_tft.script.nextion_upload:142]: Waiting for empty UART and Nextion queues [10:26:57][D][addon_upload_tft.script.nextion_upload:151]: Starting TFT upload... [10:26:57][D][nextion.upload.idf:133]: Nextion TFT upload requested [10:26:57][D][nextion.upload.idf:134]: url: https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/main/nspanel_eu.tft [10:26:57][D][nextion.upload.idf:183]: TFT file size: 7311536 [10:26:57][D][nextion.upload.idf:195]: Updating Nextion [10:27:00][D][nextion.upload.idf:223]: Upgrade response is [] [10:27:00][E][nextion.upload.idf:228]: Preparation for tft update failed 0 "" [10:27:00][D][nextion.upload.idf:259]: Restarting Nextion
Gesendet von Mail für Windows

Von: Edward Firmo
Gesendet: Mittwoch, 17. Januar 2024 09:33
An: Blackymas/NSPanel_HA_Blueprint
Cc: SeggersCoe; Comment
Betreff: Re: [Blackymas/NSPanel_HA_Blueprint] `BUG` TFT Upload with esphome4.2.2 not possible but works with 4.2.1 (Issue #1659)

So I will work in a sequence where the legacy upload is prioritized if TFT is not detected. Give me a few minutes, please.

**Comment by bluemaster27** [comment]
I dont get the mismatch notification anymore. I deleted the old automation and made a new one, as far as I know exactly the same as the previous one. But I can't swipe left to go to button page 1.

**Comment by edwardtfn** [comment]
Sorry, I'm a bit lost with all those different things here.

If it's not related to the original post "**[TFT Upload with esphome 4.2.2 not possible but works with 4.2.1](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/1659#top)**", please create a new bug report so we can track properly.

@SeggersCoe, could you please share your log again? I've tried to fix unsuccessfully. It's quite hard to read in the way it is formatted.

@bluemaster27, what do you mean by "mismatch notification"?

**Comment by bluemaster27** [comment]
Before the 4.2.3 update i got this error notification in HA

NSPanel component version mismatch There's a version mismatch between NSPanel_HA_Blueprint components. Please update all the components to the same version.

Panel: nspanel TFT: 4.1.4 ESPHome: 4.2.2 Blueprint: 4.2.2 (expected 4.2.2)

But now that's gone, so that's a good thing.

I'll make a new report for the no swipe bug.
